Source/WebKit/Platform/ios/VideoPresentationInterfaceLMK.mm

```diff
@@ -133,21 +137,39 @@ - (void)layoutSublayers
 {
     playbackSessionInterface().startObservingNowPlayingMetadata();
     [linearMediaPlayer() enterFullscreenWithCompletionHandler:makeBlockPtr([this, protectedThis = Ref { *this }, completionHandler = WTFMove(completionHandler)] (BOOL success, NSError *error) {
-        if (auto* playbackSessionModel = this->playbackSessionModel()) {
+        if (CheckedPtr playbackSessionModel = this->playbackSessionModel()) {
             playbackSessionModel->setSpatialTrackingLabel(m_spatialTrackingLabel);
             playbackSessionModel->setSoundStageSize(WebCore::AudioSessionSoundStageSize::Large);
+
+                if (CheckedPtr playbackSessionModel = playbackSessionInterface().playbackSessionModel())
+                    playableViewController().prefersAutoDimming = playbackSessionModel->prefersAutoDimming();
+
+            m_savedAutoDimmingPreference = playableViewController().prefersAutoDimming;
         }
         completionHandler(success, error);
     }).get()];
 }
 
+void VideoPresentationInterfaceLMK::prefersAutoDimmingChanged(bool prefersAutoDimming)
+{
+    if (prefersAutoDimming == m_savedAutoDimmingPreference)
+        return;
+
+    m_savedAutoDimmingPreference = prefersAutoDimming;
+    [m_playerViewController setPrefersAutoDimming:prefersAutoDimming];
+}
+
 void VideoPresentationInterfaceLMK::dismissFullscreen(bool animated, Function<void(BOOL, NSError *)>&& completionHandler)
 {
     playbackSessionInterface().stopObservingNowPlayingMetadata();
     [linearMediaPlayer() exitFullscreenWithCompletionHandler:makeBlockPtr([this, protectedThis = Ref { *this }, completionHandler = WTFMove(completionHandler)] (BOOL success, NSError *error) {
-        if (auto* playbackSessionModel = this->playbackSessionModel()) {
+        if (CheckedPtr playbackSessionModel = playbackSessionInterface().playbackSessionModel()) {
```

Comment:
I think it would have been better to do all those changes in a separate patch, so that we narrow the scope to just auto dimming.

This isn't necessary for this change